src/StringCompression.cpp

```diff
@@ -187,25 +166,12 @@ Compression::Result Compression::Extractor::ExtractZLib(ContiguousByteBufferView
 template <auto Algorithm>
 Compression::Result Compression::Extractor::Extract(const ContiguousByteBufferView Input)
 {
-	// First see if the stack buffer has enough space:
-	{
-		Result::Static Buffer;
-		size_t BytesWrittenOut;
-
-		switch (Algorithm(m_Handle, Input.data(), Input.size(), Buffer.data(), Buffer.size(), &BytesWrittenOut))
-		{
-			case LIBDEFLATE_SUCCESS: return { Buffer, BytesWrittenOut };
-			case LIBDEFLATE_INSUFFICIENT_SPACE: break;
-			default: throw std::runtime_error("Data extraction failed.");
-		}
-	}
-
-	// No it doesn't. Allocate space on the heap to write the compression result, increasing in powers of 2.
+	// Allocate space on the heap to write the compression result, increasing in powers of 2.
 
 	auto DynamicCapacity = Result::StaticCapacity * 2;
 	while (true)
 	{
-		size_t BytesWrittenOut;
+		size_t BytesWrittenOut {};
```

Comment:
This is more readable, if MSVC really requires initialization:
```suggestion
		size_t BytesWrittenOut = 0;
```

Reply:
That is useful to know.
I will use that syntax to initialize instead.